### PR TITLE
detach: allow returning json response (SC-772)

### PR DESCRIPTION
**Pull request by lucasmoura**
## Proposed Commit Message
detach: allow returning json response

Add support for a json response for the detach command

## Test Steps
Run the modified integration test

## Desired commit type

 - [ ] Squash and merge: Using the "Proposed Commit Message"
 - [ ] Create a merge commit and use "Proposed Commit Message"
 - [x] Rebase and merge, no merge commit, rely only on existing commit messages

## Checklist

 - [x] I have updated or added any unit tests accordingly
 - [x] I have updated or added any integration tests accordingly
 - [ ] I have updated or added any documentation accordingly

### CODE REVIEW

File: uaclient/tests/test_cli_disable.py [hunk 1]
```diff
@@ -100,7 +100,7 @@ def factory_side_effect(name, ent_dict=ent_dict):
         m_entitlement_factory.side_effect = factory_side_effect
 
         m_cfg = mock.Mock()
-        m_cfg.check_lock_info.return_value = (-1, "")
+        m_cfg.check_lock_info.return_value = (-2, "")
```
Comment (orndorffgrant) [inline uaclient/tests/test_cli_disable.py:103]: Curious: why is this change necessary? It's hard to tell here

**Review by orndorffgrant — APPROVED** [review]
Code looks good and includes a nice refactor of _perform_disable. Manual tests worked well. Thanks!

File: uaclient/tests/test_cli_disable.py (hunk 1 above)
Comment (lucasmoura) [reply uaclient/tests/test_cli_disable.py:103]: Good catch, I honestly think this is a mistake on my end. I will update the code

**Comment by lucasmoura** [comment]
@orndorffgrant Code updated

File: uaclient/tests/test_cli_disable.py (hunk 1 above)
Comment (orndorffgrant) [reply uaclient/tests/test_cli_disable.py]: All good thanks!
